src/chordSymbol.js

```diff
@@ -0,0 +1,593 @@
+// [VexFlow](http://vexflow.com) - Copyright (c) Mohit Muthanna 2010.
+//
+// ## Description
+//
+// This file implements chord symbols as modifiers that can be attached to
+// notes.  Chord symbols can contain multiple 'blocks' which can contain
+// text or glyphs with various positioning options.
+//
+// See `tests/chordsymbol_tests.js` for usage examples.
+
+import { Vex } from './vex';
+import { Flow } from './tables';
+import { Glyph } from './glyph';
+import { Modifier } from './modifier';
+
+// To enable logging for this class. Set `Vex.Flow.Annotation.DEBUG` to `true`.
+function L(...args) { if (ChordSymbol.DEBUG) Vex.L('Vex.Flow.ChordSymbol', args); }
+
+export class ChordSymbol extends Modifier {
+  static get CATEGORY() { return 'chordSymbol'; }
+
+  // Text annotations can be positioned and justified relative to the note.
+  static get HorizontalJustify() {
+    return {
+      LEFT: 1,
+      CENTER: 2,
+      RIGHT: 3,
+      CENTER_STEM: 4,
+    };
+  }
+
+  static get HorizontalJustifyString() {
+    return {
+      left: ChordSymbol.HorizontalJustify.LEFT,
+      right: ChordSymbol.HorizontalJustify.RIGHT,
+      center: ChordSymbol.HorizontalJustify.CENTER,
+      centerStem: ChordSymbol.HorizontalJustify.CENTER_STEM,
+    };
+  }
+
+
+  static get VerticalJustify() {
+    return {
+      TOP: 1,
+      BOTTOM: 2,
+    };
+  }
+
+  static get VerticalJustifyString() {
+    return {
+      top: ChordSymbol.VerticalJustify.TOP,
+      above: ChordSymbol.VerticalJustify.TOP,
+      below: ChordSymbol.VerticalJustify.BOTTOM,
+      bottom: ChordSymbol.VerticalJustify.BOTTOM
+    };
+  }
+
+  // Glyph data
+  static get GLYPHS() {
+    return {
+      'diminished': {
+        code: 'csymDiminished',
+        scale: 1,
+        y_offset: 0
+      },
+      'dim': {
+        code: 'csymDiminished',
+        scale: 1,
+        y_offset: 0
+      },
+      'halfDiminished': {
+        code: 'csymHalfDiminished',
+        scale: 1,
+        y_offset: 0
+      },
+      '+': {
+        code: 'csymAugmented',
+        scale: 1,
+        y_offset: 0
+      },
+      'augmented': {
+        code: 'csymAugmented',
+        scale: 1,
+        y_offset: 0
+      },
+      'majorSeventh': {
+        code: 'csymMajorSeventh',
+        scale: 1,
+        y_offset: 0
+      },
+      'minor': {
+        code: 'csymMinor',
+        scale: 1,
+        y_offset: 0
+      },
+      '-': {
+        code: 'csymMinor',
+        scale: 1,
+        y_offset: 0
+      },
+      '(': {
+        code: 'csymParensLeftTall',
+        scale: 0.8,
+        y_offset: 1
+      },
+      'leftParen': {
+        code: 'csymParensLeftTall',
+        scale: 0.8,
+        y_offset: 1
+      },
+      ')': {
+        code: 'csymParensRightTall',
+        scale: 0.8,
+        y_offset: 1
+      },
+      'rightParen': {
+        code: 'csymParensRightTall',
+        scale: 0.8,
+        y_offset: 1
```

Comment:
Can you move these `scale` and `y_offset` to the font metrics files please? We're trying to remove constants from the code.

Reply:
done